server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -558,6 +521,43 @@ public void cleanup(JsonParserIterator<T> iterFromMake)
     return retVal;
   }
 
+  private <T> void cancelQuery(Query<T> query, String cancelUrl)
+  {
+    Runnable cancelRunnable = () -> {
+      try {
+        Future<StatusResponseHolder> responseFuture = httpClient.go(
+            new Request(HttpMethod.DELETE, new URL(cancelUrl))
+            .setContent(objectMapper.writeValueAsBytes(query))
+            .setHeader(HttpHeaders.Names.CONTENT_TYPE, isSmile ? SmileMediaTypes.APPLICATION_JACKSON_SMILE : MediaType.APPLICATION_JSON),
+            StatusResponseHandler.getInstance(),
+            Duration.standardSeconds(1));
+
+        Runnable checkRunnable = () -> {
+          try {
+            if (!responseFuture.isDone()) {
+              log.error("Error cancelling query[%s]: ", query);
+            }
+            StatusResponseHolder response = responseFuture.get();
+            if (response.getStatus().getCode() >= 500) {
+              log.error("Error cancelling query[%s]: queriable node returned status[%d] [%s].",
+                  query,
+                  response.getStatus().getCode(),
+                  response.getStatus().getReasonPhrase());
+            }
+          }
+          catch (ExecutionException | InterruptedException e) {
+            log.error("Error cancelling query[%s]: ", query);
+          }
+        };
+        queryCancellationExecutor.schedule(checkRunnable, 5, TimeUnit.SECONDS);
+      }
+      catch (IOException e) {
+        log.error("Error cancelling query[%s]: ", query);
```

Comment:
Please add `e` as the first argument to `log.error()`: `log.error(e, "Error cancelling query[%s]: ", query)`